src/Razor/Microsoft.AspNetCore.Razor.Language/src/Legacy/SourceLocationTracker.cs

```diff
@@ -52,12 +52,34 @@ public static SourceLocation Advance(SourceLocation location, string text)
             return tracker.CurrentLocation;
         }
 
+        public static SourceLocation Advance(SourceLocation location, StringSegment text)
+        {
+            var tracker = new SourceLocationTracker(location);
```

Comment:
Would be nice to not have to allocate this temporary `SourceLocationTracker`

Reply:
Done. This type is no longer an instance